sky/clouds/kubernetes.py

```diff
@@ -406,3 +406,27 @@ def query_status(cls, name: str, tag_filters: Dict[str, str],
                 cluster_status.append(status_lib.ClusterStatus.INIT)
         # If pods are not found, we don't add them to the return list
         return cluster_status
+
+    @classmethod
+    def query_env_vars(cls, name: str) -> Dict[str, str]:
+        namespace = kubernetes_utils.get_current_kube_config_context_namespace()
+        pod = kubernetes.core_api().list_namespaced_pod(
+            namespace,
+            label_selector=f'skypilot-cluster={name},ray-node-type=head'
+        ).items[0]
+        response = kubernetes.stream()(
```

Comment:
Is it possible to set a timeout on this stream call (like we have `_request_timeout=kubernetes.API_TIMEOUT` in the rest of our code)?

Reply:
Yes it should be. I'll add the timeout.